external/kinesis-asl/src/main/scala/org/apache/spark/streaming/kinesis/KinesisReceiver.scala

```diff
@@ -35,10 +36,65 @@ import org.apache.spark.streaming.receiver.{BlockGenerator, BlockGeneratorListen
 import org.apache.spark.util.Utils
 
 private[kinesis]
-case class SerializableAWSCredentials(accessKeyId: String, secretKey: String)
-  extends AWSCredentials {
-  override def getAWSAccessKeyId: String = accessKeyId
-  override def getAWSSecretKey: String = secretKey
+case class SerializableKCLAuthProvider(
```

Comment:
Can we simplify this piece of code? There are too many options. Might I suggest something like:

```scala
trait SerializableAWSAuthProvider extends Serializable {
  def getProvider: AWSCredentialsProvider
}

case class BasicCredentialProvider(
    accessKeyId: String,
    secretKey: String) extends SerializableAWSAuthProvider {
  def getProvider: ...
}

case class STSCredentialsProvider(
    roleArn: String,
    sessionName: String,
    externalId: Option[String], 
    credentials: Option[BasicCredentialProvider]) ...

case object DefaultProvider extends DefaultAWSCredentialsProviderChain
```